### PR TITLE
cleanup: Refactor ObjectWriteStreambuf.

### PR DESCRIPTION
**Pull request by coryan**
There `internal::CurlObjectWriteStreambuf` was not adding value, there
was nothing "curl" about it (not anymore), it just made the inheritance
tree deeper.

Updated the API/ABI baseline, all the changes are in the `internal::`
namespace and therefore not part of the public API.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2858)

### CODE REVIEW

File: google/cloud/storage/internal/object_streambuf.h [hunk 1]
```diff
@@ -89,7 +91,13 @@ class ObjectReadStreambuf : public std::basic_streambuf<char> {
  */
 class ObjectWriteStreambuf : public std::basic_streambuf<char> {
  public:
-  ObjectWriteStreambuf() : std::basic_streambuf<char>() {}
+  ObjectWriteStreambuf() = default;
+
+  explicit ObjectWriteStreambuf(
```
Comment (devbww) [inline google/cloud/storage/internal/object_streambuf.h]: Do we bother marking multi-argument constructors as `explicit`?

File: google/cloud/storage/testbench/gcs_bucket.py [hunk 2]
```diff
@@ -601,6 +601,7 @@ def receive_upload_chunk(self, gcs_url, request):
         total = end
         final_chunk = False
         content_range = request.headers.get('content-range')
+        print("\n\n%s" % content_range)
```
Comment (devbww) [inline google/cloud/storage/testbench/gcs_bucket.py]: Testing remnant?

File: google/cloud/storage/internal/object_streambuf.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/object_streambuf.h]: Removed. We have some multi-argument constructors marked explicit and some not `¯\_(ツ)_/¯`

File: google/cloud/storage/testbench/gcs_bucket.py (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/testbench/gcs_bucket.py]: Yes. Removed.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**